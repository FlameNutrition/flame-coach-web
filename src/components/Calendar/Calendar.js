import React, { useState } from 'react';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import { NavigateNextRounded as NextMonth, NavigateBeforeRounded as PreviousMonth } from '@material-ui/icons';
import {
  Box, Card, CardContent, CardHeader, Divider, makeStyles, Typography
} from '@material-ui/core';
import CalendarReact from 'react-calendar';
// FIXME: Try to understand why the Calendar.module.css does not work
import './Calendar.css';

const useStyles = makeStyles(() => ({
  root: {
    height: '100%'
  }
}));

const Calendar = ({
  daily, onChangeCalendar
}) => {
  const classes = useStyles();
  const [calendar, setCalendar] = useState(null);

  const infoDaily = (
    <Typography component="div" align="right" variant="caption">
      <strong>Day:</strong>
      {' '}
      Double click date
    </Typography>
  );

  const infoPeriod = (
    <>
      {infoDaily}
      <Typography component="div" align="right" variant="caption">
        <strong>Period:</strong>
        {' '}
        Select a start and end date
      </Typography>
    </>
  );

  return (
    <Card
      className={clsx(classes.root)}
    >
      <CardHeader title="Calendar" />
      <Divider />
      <CardContent>
        <Box>
          <CalendarReact
            onChange={(value) => {
              setCalendar(value);
              onChangeCalendar(value);
            }}
            selectRange={!daily}
            value={calendar}
            nextLabel={<NextMonth />}
            next2Label={null}
            returnValue="range"
            prevLabel={<PreviousMonth />}
            prev2Label={null}
          />
        </Box>
        <Box>
          {!daily ? infoPeriod : null}
        </Box>
      </CardContent>
    </Card>
  );
};

Calendar.propTypes = {
  daily: PropTypes.bool,
  onChangeCalendar: PropTypes.func.isRequired
};

Calendar.defaultProps = {
  daily: false,
};
export default Calendar;
