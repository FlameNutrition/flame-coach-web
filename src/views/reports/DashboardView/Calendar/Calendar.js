import React, { useState } from 'react';
import clsx from 'clsx';
import logger from 'loglevel';
import PropTypes from 'prop-types';
import moment from 'moment';
import NextMonth from '@material-ui/icons/NavigateNextRounded';
import PreviousMonth from '@material-ui/icons/NavigateBeforeRounded';
import {
  Box, Card, CardContent, CardHeader, Divider, makeStyles
} from '@material-ui/core';
import Calendar from 'react-calendar';
// FIXME: Try to understand why the Calendar.module.css does not work
import './Calendar.css';

const useStyles = makeStyles(() => ({
  root: {
    height: '100%'
  }
}));

const TrafficByDevice = ({ className, ...rest }) => {
  const classes = useStyles();
  const [calendar, setCalendar] = useState(new Date());

  const handleChangeCalendar = (value) => {
    logger.info('Change the calendar date', moment(value));
    setCalendar(value);
  };

  return (
    <Card
      className={clsx(classes.root, className)}
      {...rest}
    >
      <CardHeader title="Calendar" />
      <Divider />
      <CardContent>
        <Box
          position="relative"
        >
          <Calendar
            onChange={handleChangeCalendar}
            value={calendar}
            nextLabel={<NextMonth />}
            next2Label={null}
            prevLabel={<PreviousMonth />}
            prev2Label={null}
          />
        </Box>
      </CardContent>
    </Card>
  );
};

TrafficByDevice.propTypes = {
  className: PropTypes.string
};

export default TrafficByDevice;
