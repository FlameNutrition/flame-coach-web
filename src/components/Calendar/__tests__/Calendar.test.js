import Calendar from '../Calendar';
import React from 'react';
import sinon from 'sinon';
import { render, screen } from '../../../testing/test-utils';

describe('<Calendar />', () => {

  const changeCalenderHandlerSpy = sinon.spy();

  afterEach(() => {
    jest.clearAllMocks();
    //cleanup();
  });

  it('Show multiple selection calendar with labels', () => {

    const { container } = render(
      <Calendar
        daily={false}
        onChangeCalendar={changeCalenderHandlerSpy}
      />
    );

    expect(container)
      .toMatchSnapshot();

    expect(screen.queryByText('Calendar'))
      .not
      .toBeNull();
    expect(screen.queryByText('Day:'))
      .not
      .toBeNull();
    expect(screen.queryByText('Double click date'))
      .not
      .toBeNull();
    expect(screen.queryByText('Period:'))
      .not
      .toBeNull();
    expect(screen.queryByText('Select a start and end date'))
      .not
      .toBeNull();

  });

  it('Show daily selection calendar', () => {

    render(
      <Calendar
        daily
        onChangeCalendar={changeCalenderHandlerSpy}
      />
    );

    expect(screen.queryByText('Day:'))
      .toBeNull();
    expect(screen.queryByText('Double click date'))
      .toBeNull();
    expect(screen.queryByText('Period:'))
      .toBeNull();
    expect(screen.queryByText('Select a start and end date'))
      .toBeNull();
  });

});
