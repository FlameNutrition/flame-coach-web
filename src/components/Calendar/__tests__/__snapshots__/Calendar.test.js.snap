// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar /> Show multiple selection calendar with labels 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardHeader-root"
    >
      <div
        class="MuiCardHeader-content"
      >
        <span
          class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
        >
          Calendar
        </span>
      </div>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiCardContent-root"
    >
      <div
        class="MuiBox-root MuiBox-root-2"
      >
        <div
          class="react-calendar react-calendar--selectRange"
        >
          <div
            class="react-calendar__navigation"
            style="display: flex;"
          >
            <button
              aria-label=""
              class="react-calendar__navigation__arrow react-calendar__navigation__prev-button"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M14.91 6.71a.9959.9959 0 00-1.41 0L8.91 11.3c-.39.39-.39 1.02 0 1.41l4.59 4.59c.39.39 1.02.39 1.41 0 .39-.39.39-1.02 0-1.41L11.03 12l3.88-3.88c.38-.39.38-1.03 0-1.41z"
                />
              </svg>
            </button>
            <button
              aria-label=""
              class="react-calendar__navigation__label"
              style="flex-grow: 1;"
              type="button"
            >
              <span
                class="react-calendar__navigation__label__labelText react-calendar__navigation__label__labelText--from"
              >
                June 2021
              </span>
            </button>
            <button
              aria-label=""
              class="react-calendar__navigation__arrow react-calendar__navigation__next-button"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9.31 6.71c-.39.39-.39 1.02 0 1.41L13.19 12l-3.88 3.88c-.39.39-.39 1.02 0 1.41.39.39 1.02.39 1.41 0l4.59-4.59c.39-.39.39-1.02 0-1.41L10.72 6.7c-.38-.38-1.02-.38-1.41.01z"
                />
              </svg>
            </button>
          </div>
          <div
            class="react-calendar__viewContainer"
          >
            <div
              class="react-calendar__month-view"
            >
              <div
                style="display: flex; align-items: flex-end;"
              >
                <div
                  style="flex-grow: 1; width: 100%;"
                >
                  <div
                    class="react-calendar__month-view__weekdays"
                    style="display: flex; flex-wrap: no-wrap;"
                  >
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    >
                      <abbr
                        aria-label="Monday"
                        title="Monday"
                      >
                        Mon
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    >
                      <abbr
                        aria-label="Tuesday"
                        title="Tuesday"
                      >
                        Tue
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    >
                      <abbr
                        aria-label="Wednesday"
                        title="Wednesday"
                      >
                        Wed
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    >
                      <abbr
                        aria-label="Thursday"
                        title="Thursday"
                      >
                        Thu
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    >
                      <abbr
                        aria-label="Friday"
                        title="Friday"
                      >
                        Fri
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    >
                      <abbr
                        aria-label="Saturday"
                        title="Saturday"
                      >
                        Sat
                      </abbr>
                    </div>
                    <div
                      class="react-calendar__month-view__weekdays__weekday"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                    >
                      <abbr
                        aria-label="Sunday"
                        title="Sunday"
                      >
                        Sun
                      </abbr>
                    </div>
                  </div>
                  <div
                    class="react-calendar__month-view__days"
                    style="display: flex; flex-wrap: wrap;"
                  >
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="May 31, 2021"
                      >
                        31
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 1, 2021"
                      >
                        1
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 2, 2021"
                      >
                        2
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 3, 2021"
                      >
                        3
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 4, 2021"
                      >
                        4
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 5, 2021"
                      >
                        5
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 6, 2021"
                      >
                        6
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 7, 2021"
                      >
                        7
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 8, 2021"
                      >
                        8
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 9, 2021"
                      >
                        9
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 10, 2021"
                      >
                        10
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 11, 2021"
                      >
                        11
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 12, 2021"
                      >
                        12
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 13, 2021"
                      >
                        13
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 14, 2021"
                      >
                        14
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 15, 2021"
                      >
                        15
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 16, 2021"
                      >
                        16
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 17, 2021"
                      >
                        17
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 18, 2021"
                      >
                        18
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 19, 2021"
                      >
                        19
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 20, 2021"
                      >
                        20
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 21, 2021"
                      >
                        21
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 22, 2021"
                      >
                        22
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 23, 2021"
                      >
                        23
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 24, 2021"
                      >
                        24
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 25, 2021"
                      >
                        25
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 26, 2021"
                      >
                        26
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__tile--now react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 27, 2021"
                      >
                        27
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 28, 2021"
                      >
                        28
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 29, 2021"
                      >
                        29
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="June 30, 2021"
                      >
                        30
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="July 1, 2021"
                      >
                        1
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--neighboringMonth"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="July 2, 2021"
                      >
                        2
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="July 3, 2021"
                      >
                        3
                      </abbr>
                    </button>
                    <button
                      class="react-calendar__tile react-calendar__month-view__days__day react-calendar__month-view__days__day--weekend react-calendar__month-view__days__day--neighboringMonth"
                      style="flex-basis: 14.285714285714286%; max-width: 14.285714285714286%; overflow: hidden;"
                      type="button"
                    >
                      <abbr
                        aria-label="July 4, 2021"
                      >
                        4
                      </abbr>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiBox-root MuiBox-root-3"
      >
        <div
          class="MuiTypography-root MuiTypography-caption MuiTypography-alignRight"
        >
          <strong>
            Day:
          </strong>
           
          Double click date
        </div>
        <div
          class="MuiTypography-root MuiTypography-caption MuiTypography-alignRight"
        >
          <strong>
            Period:
          </strong>
           
          Select a start and end date
        </div>
      </div>
    </div>
  </div>
</div>
`;
